gix/src/revision/spec/parse/delegate/revision.rs

```diff
@@ -106,12 +106,51 @@ impl delegate::Revision for Delegate<'_> {
     fn reflog(&mut self, query: ReflogLookup) -> Option<()> {
         self.unset_disambiguate_call();
         match query {
-            ReflogLookup::Date(_date) => {
-                // TODO: actually do this - this should be possible now despite incomplete date parsing
-                self.err.push(Error::Planned {
-                    dependency: "remote handling and ref-specs are fleshed out more",
-                });
-                None
+            ReflogLookup::Date(date) => {
+                let r = match &mut self.refs[self.idx] {
+                    Some(r) => r.clone().attach(self.repo),
+                    val @ None => match self.repo.head().map(crate::Head::try_into_referent) {
+                        Ok(Some(r)) => {
+                            *val = Some(r.clone().detach());
+                            r
+                        }
+                        Ok(None) => {
+                            self.err.push(Error::UnbornHeadsHaveNoRefLog);
+                            return None;
+                        }
+                        Err(err) => {
+                            self.err.push(err.into());
+                            return None;
+                        }
+                    },
+                };
```

Comment:
question since this is duplicated with the other part should I create a closure. This duplication is probably fine but it kinda bothers me.

Reply:
Refactor at will, but please do so in a separate commit.